### PR TITLE
feat(cloud): Migrate openhands cloud to V1 API endpoints

### LINKED ISSUE #629: [CLI] Migrate `openhands cloud` to use V1 API endpoints
## Summary

The `openhands cloud` command currently uses V0 (non-versioned) API endpoints for creating conversations, while OpenHands Cloud has standardized V1 API endpoints available. This creates inconsistency and potential compatibility issues.

## Current Behavior

When executing `openhands cloud -t "here is my prompt"`, the CLI uses:

- `POST /api/conversations` (V0 endpoint) - for creating conversations
- `GET /api/user/info` (V0 endpoint) - for user information
- `GET /api/keys/llm/byor` (V0 endpoint) - for LLM API key
- `GET /api/settings` (V0 endpoint) - for user settings

The only V1 endpoint currently used is:
- `GET /api/v1/app-conversations?ids={conversation_id}` - for fetching conversation info

## Expected Behavior

The CLI should use the standardized V1 API endpoints:

- `POST /api/v1/app-conversations` - for creating conversations
- Potentially migrate other endpoints to V1 equivalents if available

### Current Implementation

The conversation creation happens in:
- **File**: `openhands_cli/auth/api_client.py` (line 80)
- **Method**: `create_conversation()`
```python
async def create_conversation(
    self, json_data: dict[str, Any] | None = None
) -> httpx.Response:
    return await self.post("/api/conversations", self._headers, json_data)
```

### V1 API Specification

According to the OpenHands API documentation, the V1 endpoint should be:
- **Endpoint**: `POST /api/v1/app-conversations`
- **Auth**: Bearer token in Authorization header
- **Payload**:
  ```json
  {
    "initial_message": {
      "content": [{"type": "text", "text": "..."}]
    },
    "selected_repository": "owner/repo",
    "selected_branch": "branch-name"
  }
  ```

### Key Differences

1. **Endpoint path**: `/api/conversations` → `/api/v1/app-conversations`
2. **Payload structure**: V1 uses a structured `initial_message` object with content array, not just `initial_user_msg` string
3. **Response**: V1 may return async start-task data that needs polling

## Proposed Changes

1. Update `OpenHandsApiClient.create_conversation()` to use `/api/v1/app-conversations`
2. Update payload format in `openhands_cli/cloud/conversation.py` to match V1 spec:
   - Transform `initial_user_msg` to structured `initial_message` format
   - Update repository field naming if needed (`repository` → `selected_repository`)
3. Handle potential async start-task response (poll for `app_conversation_id` if needed)
4. Update tests in `tests/cloud/test_conversation.py` to reflect new payload structure
5. Consider migrating other V0 endpoints to V1 equivalents

## Related Files

- `openhands_cli/cloud/conversation.py` - Main conversation creation logic
- `openhands_cli/auth/api_client.py` - API client implementation
- `openhands_cli/cloud/command.py` - Cloud command handler
- `tests/cloud/test_conversation.py` - Tests for cloud conversation functionality

## Benefits

- **Consistency**: Aligns with OpenHands Cloud's standardized API
- **Future-proofing**: V1 APIs are versioned and more stable
- **Feature parity**: Access to newer V1-only features
- **Maintainability**: Easier to track API changes and deprecations

## References

- OpenHands API V1 documentation: See `openhands-api` skill
- Current CLI implementation: `openhands_cli/cloud/conversation.py`

### PR DESCRIPTION
**Pull request by sjathin**
## Summary of PR

Migrates openhands cloud from the deprecated V0 API (/api/conversations) to the V1 API (/api/v1/app-conversations), per the [Cloud API migration guide](https://docs.all-hands.dev/usage/cloud/cloud-api#migrating-
from-v0-to-v1-api).

API changes:
- Endpoint: /api/conversations → /api/v1/app-conversations
- Payload: initial_user_msg (string) → initial_message.content (array of content objects)
- Field: repository → selected_repository
- Response: V1 returns an async start-task; polls /api/v1/app-conversations/start-tasks?ids= for app_conversation_id

Code changes:
- openhands_cli/auth/api_client.py: Updated create_conversation endpoint, added get_start_task_status method, extracted _v1_conversations_path helper to deduplicate path building with get_conversation_info
- openhands_cli/cloud/conversation.py: Updated payload format, added polling loop for app_conversation_id with error handling for start-task ERROR status
- tests/cloud/test_conversation.py: Updated existing tests for V1 payload, added 5 new tests covering polling success, timeout, error, null response, and missing task ID

## Demo Screenshots/Videos

Manual test output:

```
▶ uv run openhands cloud -t "here is my prompt"
zsh: correct 'openhands' to '.openhands' [nyae]? n
Detected repository: sjathin/OpenHands-CLI
Detected branch: open-hands-cli-629
Creating cloud conversation...
Waiting for conversation to start...
Conversation ID: a824ad8bc0e64d54b8084f731095e824
View in browser: https://app.all-hands.dev/conversations/a824ad8bc0e64d54b8084f731095e824
Cloud conversation created successfully! 🚀
```

https://app.all-hands.dev/conversations/a824ad8bc0e64d54b8084f731095e824

## Change Type

- [ ] Bug fix
- [x] New feature
- [ ] Breaking change
- [ ] Refactor
- [ ] Other (dependency update, docs, typo fixes, etc.)

## Checklist

- [x] I have read and reviewed the code and I understand what the code is doing.
- [x] I have tested the code to the best of my ability and ensured it works as expected.

## Fixes

Resolves #629 

## Release Notes

- [x] Include this change in the Release Notes.

openhands cloud now uses the V1 API for creating conversations, with structured message format and async start-task polling. The deprecated V0 endpoint (/api/conversations) is no longer used.

### CODE REVIEW

File: openhands_cli/cloud/conversation.py [hunk 1]
```diff
@@ -77,7 +85,29 @@ async def create_cloud_conversation(
...
+                app_conversation_id = task_info.get("app_conversation_id")
+                if app_conversation_id:
+                    break
+                if status == "ERROR":
+                    detail = task_info.get("detail", "Unknown error")
+                    raise CloudConversationError(
+                        f"Conversation failed to start: {detail}"
+                    )
+
+    conversation_id = app_conversation_id or task_id
```
Comment (jpshackelford) [inline openhands_cli/cloud/conversation.py:114]: **Suggestion:** If polling exhausts all attempts without getting `app_conversation_id`, the user gets a URL that may not work immediately—with no indication anything is amiss.

Since we've made it this far, we know authentication and configuration are fine; it's just that the backend is slow. Consider adding a warning before falling back:

```python
if not app_conversation_id:
    console_print(
        "⚠️ Conversation is still initializing. The link below may take a moment to become active.",
        style=OPENHANDS_THEME.warning,
    )
```

This sets user expectations without requiring any retry logic.

File: openhands_cli/cloud/conversation.py [hunk 2]
```diff
@@ -24,6 +25,9 @@
 
 logger = logging.getLogger(__name__)
 
+_POLL_INTERVAL_SECONDS = 2
+_POLL_MAX_ATTEMPTS = 15
```
Comment (jpshackelford) [inline openhands_cli/cloud/conversation.py]: **Suggestion:** Patching module-level constants in tests is fragile—if someone imports these differently, the patch breaks silently.

Consider adding these as keyword-only parameters with defaults:

```python
async def create_cloud_conversation(
    server_url: str,
    api_key: str,
    initial_user_msg: str,
    *,
    poll_interval: float = 2.0,
    poll_max_attempts: int = 15,
) -> dict[str, Any]:
```

Then tests can call directly with `poll_interval=0.01, poll_max_attempts=2` without any patching. Production callers are unaffected since the defaults match current behavior.

File: tests/cloud/test_conversation.py [hunk 3]
```diff
@@ -265,3 +274,136 @@ async def test_create_cloud_conversation_propagates_api_error_as_cloud_error():
             CloudConversationError, match=r"Failed to create conversation: boom"
         ):
             await create_cloud_conversation("https://server", "key", "hi")
+
```
Comment (jpshackelford) [inline tests/cloud/test_conversation.py:277]: **Nice to have (not required for this PR):** The new tests mock `OpenHandsApiClient` at the class level, so the real client code never runs. This verifies that methods were *called*, but not that they *work*.

If you have time, consider following the fixture pattern from [jpshackelford/oh-utils](https://github.com/jpshackelford/oh-utils/tree/main/tests/fixtures)—mock HTTP at the transport layer with `respx` and let the real `OpenHandsApiClient` execute. This would catch bugs in path building, response parsing, and error handling.

Example fixtures to create:
- `start_task_pending.json` — returns `{"status": "WORKING", "app_conversation_id": null}`
- `start_task_ready.json` — returns `{"status": "READY", "app_conversation_id": "ac1"}`
- `start_task_error.json` — returns `{"status": "ERROR", "detail": "sandbox failed"}`

Not blocking—the current tests are fine for this PR—but would strengthen the test suite for future changes.

File: openhands_cli/auth/api_client.py [hunk 4]
```diff
@@ -105,10 +106,25 @@ async def get_conversation_info(
             raise ApiClientError(f"Request to {path!r} failed: {e}") from e
 
         data: list[dict[str, Any]] = response.json()
-        if len(data) > 0:
+        if data and data[0]:
             return data[0]
         return None
 
+    @staticmethod
```
Comment (jpshackelford) [inline openhands_cli/auth/api_client.py:113]: **Nice to have:** A quick unit test for this helper would document expected behavior and catch any future typos:

```python
def test_v1_conversations_path_without_endpoint():
    assert OpenHandsApiClient._v1_conversations_path("abc") == "/api/v1/app-conversations?ids=abc"

def test_v1_conversations_path_with_start_tasks():
    assert OpenHandsApiClient._v1_conversations_path("abc", "start-tasks") == "/api/v1/app-conversations/start-tasks?ids=abc"
```

Low priority—the method is simple and integration tests would catch issues—but it's a 30-second addition.

File: openhands_cli/cloud/conversation.py (hunk 1 above)
Comment (sjathin) [reply openhands_cli/cloud/conversation.py:114]: Nice catch! Agreed — added a warning when polling exhausts all attempts. Also added a test to verify the warning is printed.

File: openhands_cli/cloud/conversation.py (hunk 2 above)
Comment (sjathin) [reply openhands_cli/cloud/conversation.py]: Done — moved to keyword-only params with defaults (poll_interval=2.0, poll_max_attempts=15). Tests now pass values directly instead of patching module constants.

File: tests/cloud/test_conversation.py (hunk 3 above)
Comment (sjathin) [reply tests/cloud/test_conversation.py:277]: Done — added respx as a dev dependency and created tests/cloud/test_api_client_transport.py with 8 transport-level tests that exercise the real OpenHandsApiClient code. JSON mock responses live in
tests/cloud/mock_responses/.

File: openhands_cli/auth/api_client.py (hunk 4 above)
Comment (sjathin) [reply openhands_cli/auth/api_client.py:113]: Done

**Comment by jpshackelford** [comment]
## Manual Testing Results

_This comment was created by an AI assistant (OpenHands) on behalf of @jpshackelford._

### Automated Tests: ✅ All 48 tests pass

```
tests/cloud/test_conversation.py - 23 passed
tests/cloud/test_api_client_transport.py - 8 passed  
tests/auth/test_api_client.py - 17 passed
```

### Manual Testing: ✅ Both scenarios work

**1. With repo context** (from within the OpenHands-CLI repo on the PR branch):
```
$ uv run openhands cloud -t "Test from CLI - please echo hello and close"

Detected repository: OpenHands/OpenHands-CLI
Detected branch: open-hands-cli-629
Creating cloud conversation...
Waiting for conversation to start...
Conversation ID: ee22dbb6ffc94fb0bfa1a39231337893
View in browser: https://app.all-hands.dev/conversations/ee22dbb6ffc94fb0bfa1a39231337893
Cloud conversation created successfully! 🚀
```

Verified via V1 API - `selected_repository` and `selected_branch` correctly populated:
```json
{
    "id": "ee22dbb6ffc94fb0bfa1a39231337893",
    "selected_repository": "OpenHands/OpenHands-CLI",
    "selected_branch": "open-hands-cli-629",
    ...
}
```

**2. Without repo context** (from `/tmp`):
```
$ openhands cloud -t "Test without repo context - simple hello"

Creating cloud conversation...
Waiting for conversation to start...
Conversation ID: 85cb22a6cd17481dba95d99e07994ca7
View in browser: https://app.all-hands.dev/conversations/85cb22a6cd17481dba95d99e07994ca7
Cloud conversation created successfully! 🚀
```

Verified via V1 API - fields are null as expected:
```json
{
    "id": "85cb22a6cd17481dba95d99e07994ca7",
    "selected_repository": null,
    "selected_branch": null,
    ...
}
```

### Code Review Notes

- ✅ Clean migration following the V0→V1 migration guide
- ✅ Configurable polling parameters (`poll_interval`, `poll_max_attempts`)
- ✅ Error handling for "ERROR" status from start-task
- ✅ Graceful fallback with warning if polling times out
- ✅ Good test coverage including edge cases
- ✅ DRYs up path building with `_v1_conversations_path` helper

**LGTM** 👍

**Review by jpshackelford — APPROVED** [review]
@sjathin Thanks again for the great work on this!

**Comment by jpshackelford** [comment]
Squash merging this PR.

_This action was performed by an AI assistant (OpenHands) on behalf of @jpshackelford._
